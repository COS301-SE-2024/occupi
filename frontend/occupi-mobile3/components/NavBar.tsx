import React from 'react';
import {  StyleSheet } from 'react-native';
import { Text, View, Button, Icon, SearchIcon, CalendarDaysIcon, BellIcon } from '@gluestack-ui/themed';
import StyledExpoRouterLink from './StyledExpoRouterLink';
import { Feather } from '@expo/vector-icons';
import { FontAwesome6 } from '@expo/vector-icons';
import { router } from 'expo-router';
import {
    Avatar,
    AvatarBadge,
    AvatarFallbackText,
    AvatarImage,
  } from "@gluestack-ui/themed"
import { BlurView } from 'expo-blur';
import { Appearance, useColorScheme } from 'react-native';

const NavBar = () => {
  let colorScheme = useColorScheme();
  const styles = getStyles(colorScheme);
  return (
    <BlurView tint="light" pb="$8" backgroundColor={colorScheme === 'dark' ? 'black' : '#fff'} intensity={20} style={styles.container}>  
        <Button onPress={() => router.push('/home')} flex="$1" mt="$2" w="$22" backgroundColor="$none" title="Home"  flexDirection="$column"><Feather name="home" size={24} color={colorScheme === 'dark' ? 'white' : 'black'} /><Text numberOfLines={1} w="$9" fontSize="$12" color={colorScheme === 'dark' ? 'white' : 'black'}>Home</Text></Button>
        <Button onPress={() => router.push('/bookings')} flex="$1" mt="$2" w="$22" backgroundColor="$none" title="Search"  flexDirection="$column"><Icon as={SearchIcon} w="$6" h="$6" color={colorScheme === 'dark' ? 'white' : 'black'} /><Text numberOfLines={1} w="$10" fontSize="$12" color={colorScheme === 'dark' ? 'white' : 'black'}>Search</Text></Button>
        <Button onPress={() => router.push('/bookings')} flex="$1" mt="$2" w="$22" backgroundColor="$none" title="Book"  flexDirection="$column"><Icon as={CalendarDaysIcon} w="$6" h="$6" color={colorScheme === 'dark' ? 'white' : 'black'} /><Text numberOfLines={1} numberOfLines={1} fontSize="$12" color={colorScheme === 'dark' ? 'white' : 'black'}>Book</Text></Button>
        <Button onPress={() => router.push('/bookings')} flex="$1" mt="$2" w="$22" backgroundColor="$none" title="Notifications"  flexDirection="$column"><Icon as={BellIcon} w="$6" h="$6" color={colorScheme === 'dark' ? 'white' : 'black'} /><Text pl="$1" numberOfLines={1} w="$20" fontSize="$12" color={colorScheme === 'dark' ? 'white' : 'black'}>Notifications</Text></Button>
        <Button onPress={() => router.push('/settings')} flex="$1" mt="$2" w="$22" backgroundColor="$none" title="Profile"  flexDirection="$column"><FontAwesome6 name="user" size={24} color="black" color={colorScheme === 'dark' ? 'white' : 'black'} /><Text pl="$1" numberOfLines={1} w="$12" fontSize="$12" color={colorScheme === 'dark' ? 'white' : 'black'}>Profile</Text></Button>
        {/* <Avatar bgColor="$amber600" size="md" borderRadius="$full">
            <AvatarFallbackText>Tinashe Austin</AvatarFallbackText>
        </Avatar> */}
    </BlurView>
  );
};

const getStyles = (colorScheme) => StyleSheet.create({
  container: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    paddingHorizontal: 16,
    paddingBottom: 26,
    flexDirection: 'row',
    justifyContent: 'space-around',
    backgroundColor: colorScheme === 'dark' ? 'rgba(0, 0, 0, 0.8)' : 'rgba(255, 255, 255, 0.5)',
    // backgroundColor: '#fff',
    paddingVertical: 10,
    borderTopWidth: 1,
    // borderLeftWidth: 1,
    // borderRightWidth: 1,
    // borderRadius: 20,
    borderTopColor: colorScheme === 'dark' ? '#444' : '#ccc',
    borderLeftColor: '#ccc',
    borderRightColor: '#ccc',
  }
});

export default NavBar;
