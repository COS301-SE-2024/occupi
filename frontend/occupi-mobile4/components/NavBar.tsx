import React from 'react';
import { StyleSheet } from 'react-native';
import { Text, Button, Icon, SearchIcon, CalendarDaysIcon, BellIcon } from '@gluestack-ui/themed';
import { Feather } from '@expo/vector-icons';
import { FontAwesome6 } from '@expo/vector-icons';
import { router } from 'expo-router';
import { BlurView } from 'expo-blur';
import { useColorScheme } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';

const NavBar = () => {
  let colorScheme = useColorScheme();
  const styles = getStyles(colorScheme);
  return (
    <BlurView tint="light" pb={hp('3%')} backgroundColor={colorScheme === 'dark' ? 'black' : '#fff'} intensity={20} style={styles.container}>  
      <Button onPress={() => router.push('/home')} flex={1} mt={hp('1%')} w={wp('20%')} backgroundColor="none" title="Home" flexDirection="column">
        <Feather name="home" size={hp('3%')} color={colorScheme === 'dark' ? 'white' : 'black'} />
        <Text numberOfLines={1} w={wp('9%')} fontSize={wp('3%')} color={colorScheme === 'dark' ? 'white' : 'black'}>Home</Text>
      </Button>
      <Button onPress={() => router.push('/viewbookings')} flex={1} mt={hp('1%')} w={wp('20%')} backgroundColor="none" title="Search" flexDirection="column">
        <Icon as={SearchIcon} w={hp('3%')} h={hp('3%')} color={colorScheme === 'dark' ? 'white' : 'black'} />
        <Text numberOfLines={1} w={wp('20%')} fontSize={wp('3%')} color={colorScheme === 'dark' ? 'white' : 'black'}>My bookings</Text>
      </Button>
      <Button onPress={() => router.push('/bookings')} flex={1} mt={hp('1%')} w={wp('20%')} backgroundColor="none" title="Book" flexDirection="column">
        <Icon as={CalendarDaysIcon} w={hp('3%')} h={hp('3%')} color={colorScheme === 'dark' ? 'white' : 'black'} />
        <Text numberOfLines={1} w={wp('7.4%')} fontSize={wp('3%')} color={colorScheme === 'dark' ? 'white' : 'black'}>Book</Text>
      </Button>
      <Button onPress={() => router.push('/bookings')} flex={1} mt={hp('1%')} w={wp('20%')} backgroundColor="none" title="Notifications" flexDirection="column">
        <Icon as={BellIcon} w={hp('3%')} h={hp('3%')} color={colorScheme === 'dark' ? 'white' : 'black'} />
        <Text pl={wp('1%')} numberOfLines={1} w={wp('20%')} fontSize={wp('3%')} color={colorScheme === 'dark' ? 'white' : 'black'}>Notifications</Text>
      </Button>
      <Button onPress={() => router.push('/settings')} flex={1} mt={hp('1%')} w={wp('20%')} backgroundColor="none" title="Profile" flexDirection="column">
        <FontAwesome6 name="user" size={hp('3%')} color={colorScheme === 'dark' ? 'white' : 'black'} />
        <Text pl={wp('1%')} numberOfLines={1} w={wp('12%')} fontSize={wp('3%')} color={colorScheme === 'dark' ? 'white' : 'black'}>Profile</Text>
      </Button>
    </BlurView>
  );
};

const getStyles = (colorScheme) => StyleSheet.create({
  container: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    paddingHorizontal: wp('4%'),
    paddingBottom: hp('3%'),
    flexDirection: 'row',
    justifyContent: 'space-around',
    backgroundColor: colorScheme === 'dark' ? 'rgba(0, 0, 0, 0.8)' : 'rgba(255, 255, 255, 0.5)',
    paddingVertical: hp('1%'),
    borderTopWidth: 1,
    borderTopColor: colorScheme === 'dark' ? '#444' : '#ccc',
    borderLeftColor: '#ccc',
    borderRightColor: '#ccc',
  }
});

export default NavBar;
