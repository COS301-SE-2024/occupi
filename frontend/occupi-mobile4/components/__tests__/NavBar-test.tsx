import * as React from 'react';
import renderer from 'react-test-renderer';
import NavBar from '../NavBar';

it(`renders correctly`, () => {
  const tree = renderer.create(<NavBar>Snapshot test!</NavBar>).toJSON();

  expect(tree).toMatchSnapshot();
});
