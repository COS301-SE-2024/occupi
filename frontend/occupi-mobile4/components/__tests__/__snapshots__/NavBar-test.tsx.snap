// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  backgroundColor="#fff"
  pb={40}
  style={
    [
      {
        "backgroundColor": "transparent",
      },
      {
        "backgroundColor": "rgba(255, 255, 255, 0.5)",
        "borderLeftColor": "#ccc",
        "borderRightColor": "#ccc",
        "borderTopColor": "#ccc",
        "borderTopWidth": 1,
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "left": 0,
        "paddingBottom": 40,
        "paddingHorizontal": 30,
        "paddingVertical": 13.5,
        "position": "absolute",
        "right": 0,
      },
    ]
  }
>
  <ViewManagerAdapter_ExpoBlurView
    blurReductionFactor={4}
    experimentalBlurMethod="none"
    intensity={20}
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    tint="light"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    data-style=""
    dataSet={
      {
        "active": "false",
        "disabled": "false",
        "focus": "false",
        "focusVisible": "false",
        "hover": "false",
        "style": "",
      }
    }
    focusable={true}
    mt={13.5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    style={
      [
        undefined,
        {},
        {
          "backgroundColor": "none",
          "flex": 1,
          "flexDirection": "column",
        },
        undefined,
      ]
    }
    title="Home"
    w={150}
  >
    <Text />
    <Text
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      numberOfLines={1}
      style={
        [
          undefined,
          {},
          {
            "color": "black",
            "fontSize": 22.5,
          },
          undefined,
        ]
      }
      w={67.5}
    >
      Home
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    data-style=""
    dataSet={
      {
        "active": "false",
        "disabled": "false",
        "focus": "false",
        "focusVisible": "false",
        "hover": "false",
        "style": "",
      }
    }
    focusable={true}
    mt={13.5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    style={
      [
        undefined,
        {},
        {
          "backgroundColor": "none",
          "flex": 1,
          "flexDirection": "column",
        },
        undefined,
      ]
    }
    title="Search"
    w={150}
  >
    <Text />
    <Text
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      numberOfLines={1}
      style={
        [
          undefined,
          {},
          {
            "color": "black",
            "fontSize": 22.5,
          },
          undefined,
        ]
      }
      w={142.5}
    >
      My bookings
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    data-style=""
    dataSet={
      {
        "active": "false",
        "disabled": "false",
        "focus": "false",
        "focusVisible": "false",
        "hover": "false",
        "style": "",
      }
    }
    focusable={true}
    mt={13.5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    style={
      [
        undefined,
        {},
        {
          "backgroundColor": "none",
          "flex": 1,
          "flexDirection": "column",
        },
        undefined,
      ]
    }
    title="Book"
    w={150}
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="100%"
      bbWidth="100%"
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      focusable={false}
      h={40}
      meetOrSlice={0}
      minX={0}
      minY={0}
      role="img"
      stroke="currentColor"
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            undefined,
            {},
            undefined,
            {},
            undefined,
            {},
            undefined,
            {},
            {
              "color": "black",
            },
            undefined,
          ],
          {
            "flex": 0,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
      tintColor="black"
      vbHeight={24}
      vbWidth={24}
      w={40}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGPath
            d="M19 4H5C3.89543 4 3 4.89543 3 6V20C3 21.1046 3.89543 22 5 22H19C20.1046 22 21 21.1046 21 20V6C21 4.89543 20.1046 4 19 4Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M16 2V6"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M8 2V6"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M3 10H21"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M8 14H8.01"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M12 14H12.01"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M16 14H16.01"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M8 18H8.01"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M12 18H12.01"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M16 18H16.01"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      numberOfLines={1}
      style={
        [
          undefined,
          {},
          {
            "color": "black",
            "fontSize": 22.5,
          },
          undefined,
        ]
      }
      w={55.5}
    >
      Book
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    data-style=""
    dataSet={
      {
        "active": "false",
        "disabled": "false",
        "focus": "false",
        "focusVisible": "false",
        "hover": "false",
        "style": "",
      }
    }
    focusable={true}
    mt={13.5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    style={
      [
        undefined,
        {},
        {
          "backgroundColor": "none",
          "flex": 1,
          "flexDirection": "column",
        },
        undefined,
      ]
    }
    title="Notifications"
    w={150}
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="100%"
      bbWidth="100%"
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      focusable={false}
      h={40}
      meetOrSlice={0}
      minX={0}
      minY={0}
      role="img"
      stroke="currentColor"
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            undefined,
            {},
            undefined,
            {},
            undefined,
            {},
            undefined,
            {},
            {
              "color": "black",
            },
            undefined,
          ],
          {
            "flex": 0,
            "height": "100%",
            "width": "100%",
          },
        ]
      }
      tintColor="black"
      vbHeight={24}
      vbWidth={24}
      w={40}
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGPath
            d="M18 8C18 6.4087 17.3679 4.88258 16.2426 3.75736C15.1174 2.63214 13.5913 2 12 2C10.4087 2 8.88258 2.63214 7.75736 3.75736C6.63214 4.88258 6 6.4087 6 8C6 15 3 17 3 17H21C21 17 18 15 18 8Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
          <RNSVGPath
            d="M13.73 21C13.5542 21.3031 13.3018 21.5547 12.9982 21.7295C12.6946 21.9044 12.3504 21.9965 12 21.9965C11.6496 21.9965 11.3054 21.9044 11.0018 21.7295C10.6982 21.5547 10.4458 21.3031 10.27 21"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="2"
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      numberOfLines={1}
      pl={7.5}
      style={
        [
          undefined,
          {},
          {
            "color": "black",
            "fontSize": 22.5,
          },
          undefined,
        ]
      }
      w={150}
    >
      Notifications
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    data-style=""
    dataSet={
      {
        "active": "false",
        "disabled": "false",
        "focus": "false",
        "focusVisible": "false",
        "hover": "false",
        "style": "",
      }
    }
    focusable={true}
    mt={13.5}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="button"
    style={
      [
        undefined,
        {},
        {
          "backgroundColor": "none",
          "flex": 1,
          "flexDirection": "column",
        },
        undefined,
      ]
    }
    title="Profile"
    w={150}
  >
    <Text />
    <Text
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      numberOfLines={1}
      pl={7.5}
      style={
        [
          undefined,
          {},
          {
            "color": "black",
            "fontSize": 22.5,
          },
          undefined,
        ]
      }
      w={90}
    >
      Profile
    </Text>
  </View>
</View>
`;
